add info on db drivers (#1592)
strapi/documentation: docusaurus/docs/dev-docs/configurations/database.md

| `pool`<br /><br />_Optional_                             | [Database pooling options](#database-pooling-options)                                                 | `Object`  | -       |
| `acquireConnectionTimeout`<br /><br />_Optional_         | How long knex will wait before throwing a timeout error when acquiring a connection (in milliseconds) | `Integer` | `60000` |

::: note

> The `client` value may be replaced by Strapi before being passed to Knex, based on which database driver packages are available in the project

Reviewer: ```suggestion
The `client` value may be replaced by Strapi before being passed to Knex, based on which database driver packages are available in the project.
```

> A client value of sqlite will be modified by Strapi to be better-sqlite3 if the package is available in your project, or sqlite3 if it is not.

Author: ```suggestion
A `client` value of 'sqlite' will be modified by Strapi to be 'better-sqlite3' if the package is available in your project, or 'sqlite3' if it is not.
```

> The `client` value may be replaced by Strapi before being passed to Knex, based on which database driver packages are available in the project.

Author: Strapi treats `postgres`, `sqlite`, and `mysql` as generic names for the client you are using, and may replace the `client` value with the best available package before passing it on to Knex if a preferred package is available.

For example, if `better-sqlite3` is installed in your project, Strapi will replace the client value `sqlite` with `better-sqlite3` before creating the Knex connection.

> The `client` value may be replaced by Strapi before being passed to Knex, based on which database driver packages are available in the project.

Author: ```suggestion
A client value of sqlite will be modified by Strapi to be better-sqlite3 if the package is available in your project, or sqlite3 if it is not.
```

```

:::

### Alternative database driver packages

> In addition to the `client` values of '[postgres](https://www.npmjs.com/package/pg)', 'sqlite', and '[mysql](https://www.npmjs.com/package/mysql)', Strapi also allows a `client` value of '[mysql2](https://www.npmjs.com/package/mysql2)' for those who install and wish to use that package.
> 

Author: ```suggestion
In addition to `client` values of '[postgres](https://www.npmjs.com/package/pg)', 'sqlite', and '[mysql](https://www.npmjs.com/package/mysql)', Strapi also allows a `client` value of '[mysql2](https://www.npmjs.com/package/mysql2)' for those who install and wish to use that package.
```

```

:::

### Alternative database drivers

Strapi attempts to load database drivers for each client based on which packages have been added to the project.

> | client   | order of driver selected when available  |

Author: ```suggestion
| client   | order of package selected when available  |
```
I think this might sound better, but I've been going back and forth on package vs driver I'm not sure anymore. Should it be "driver package"?

| -------- | ---------------------------------------- |

> | mysql    | mysql2, mysql                            |

Author: Would it be worth linking to the packages on github or npm like we do with knex?

Reviewer: I think so, yes! :)

> | mysql    | [mysql2](https://www.npmjs.com/package/mysql2), [mysql](https://www.npmjs.com/package/mysql)                     |

Author: ```suggestion
| mysql    | [mysql](https://www.npmjs.com/package/mysql), [mysql2](https://www.npmjs.com/package/mysql2)                     |
```

> | mysql    | [mysql](https://www.npmjs.com/package/mysql), [mysql2](https://www.npmjs.com/package/mysql2)                     |

Author: ```suggestion
| mysql    | [mysql2](https://www.npmjs.com/package/mysql2), [mysql](https://www.npmjs.com/package/mysql)                     |
```
